frontend/heir/mlir_emitter.py

```diff
@@ -476,40 +476,43 @@ def emit_ext_if_needed(self, lhs, rhs):
     )
 
     if lhs_type.bitwidth > rhs_type.bitwidth:
-      return self.get_name(lhs), tmp, ext
-    return tmp, self.get_name(rhs), ext
+      return self.get_name(lhs), tmp, ext, self.typemap.get(str(long))
+    return tmp, self.get_name(rhs), ext, self.typemap.get(str(long))
```

Comment:
Why do you hardcode the returned type here? Shouldn't this just be `lhs_type` in the true case, and `rhs_type` in the false case?

Reply:
Ah, I thought it was easier to put the long type there since we want to emit the long type for the binary op.

A few lines earlier, lhs and rhs are appropriately places in short/long so what you believe should be happening is indeed what it's doing.

Reply:
addressed! in hindsight it is nicer to just use lhs_type/rhs_type